### ISSUE TITLE
One-time physics impulse adds permanent rotational force (when no applied position is provided)

### ISSUE THREAD

**Issue by tristanbob** [4ian/GDevelop#3384]
## Describe the bug

One-time physics impulse adds permanent rotational force (when no applied position is provided)

## Video

https://user-images.githubusercontent.com/8879811/146694940-d9ee075a-194f-4a7d-8605-6258a348b100.mp4

## To Reproduce
1. Create an action to apply an "impulse towards an angle" one-time only
2. Leave the fields "where to apply force" blank (I assume this uses the origin)
3. Object will move at a linear rate, but rotation speed will increase over time (like a rotation force was applied)

## Other details
* Windows 10
* GDevelop 5.0.124

**Comment by ClementPasteau** [comment]
Interesting.
When the fields X & Y are left blank, the value (0,0) is taken, which I assume is the top left corner of the canvas.
This may explain why this weird behaviour?
https://github.com/4ian/GDevelop/blob/b3e0540fedc5b5f3260d2981ef9c445da4d86115/Extensions/Physics2Behavior/physics2runtimebehavior.ts#L1386-L1404

Maybe a better behaviour would be to detect when X&Y are empty and use `ApplyLinearImpulseToCenter()` instead.
(cf doc https://box2d.org/documentation/classb2_body.html#afa249d2fc11735985211e47c3d8e16fb)

I assume the same is happening for other "apply forces" of the physics 2 behaviour

**Comment by tristanbob** [comment]
There are two (unrelated) topics:

1) You bring up an excellent point that I have wanted since the first time I used the physics engine:  Forces should automatically apply to the center unless explicitly specified.   This would be a great improvement for users.

2) A one-time action should NOT cause a constantly increasing force. There is a bug in there somewhere.

**Comment by arthuro555** [comment]
> A one-time action should NOT cause a constantly increasing rotational force. There is a bug in there somewhere.

I don't think that this is surprising? The object goes on with a constant speed forever, so the impulse is permanent. If the impulse is not applied at the center, it is to be expected that it will be rotated forever as well.

**Comment by tristanbob** [comment]
> > A one-time action should NOT cause a constantly increasing rotational force. There is a bug in there somewhere.
> 
> I don't think that this is surprising? The object goes on with a constant speed forever, so the impulse is permanent. If the impulse is not applied at the center, it is to be expected that it will be rotated forever as well.

The way I understand this is that an "impulse" is like hitting an object with a hammer.  It pushes the object for an instant but does not affect the object again.
